diff --git a/index.d.ts b/index.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..182b7bbc45c3dd685763227d5d8e2e1c95678a3a
--- /dev/null
+++ b/index.d.ts
@@ -0,0 +1,21 @@
+export type ErrorLocation = {
+  fileName: string;
+  lineNumber: number;
+  colNumber?: number;
+};
+
+type RuntimeReportingOptions = {
+  onError?: () => void;
+  filename?: string;
+};
+
+type EditorHandler = (errorLoc: ErrorLocation) => void;
+
+export const setEditorHandler: (handler: EditorHandler | null) => void;
+export const reportBuildError: (error: string) => void;
+export const reportRuntimeError: (error: Error, options: RuntimeReportingOptions) =>  void;
+export const dismissBuildError: () => void;
+export const startReportingRuntimeErrors: (options: RuntimeReportingOptions) => void;
+export const dismissRuntimeErrors: () => void;
+export const stopReportingRuntimeErrors: () => void;
+
