diff --git a/node_modules/detect-port-alt/index.d.ts b/node_modules/detect-port-alt/index.d.ts
new file mode 100644
index 0000000..982cba2
--- /dev/null
+++ b/node_modules/detect-port-alt/index.d.ts
@@ -0,0 +1,4 @@
+declare function detectPort(defaultPort: number, host: string, ): Promise<number>;
+declare function detectPort(defaultPort: number, host: string, callback: (err: any, newPort: number) => any): void;
+
+export = detectPort;
\ No newline at end of file
